src/Extension/FluentExtension.php

```diff
@@ -1021,21 +1052,25 @@ public function updateFluentCMSField(FormField $field)
     /**
      * Require that this record is saved in the given locale for it to be visible
      *
-     * @return bool
+     * @return string
      */
-    protected function requireSavedInLocale()
+    protected function getInheritanceMode(): string
     {
-        if (FluentState::singleton()->getIsFrontend()) {
-            return $this->owner->config()->get('frontend_publish_required');
-        }
-
-        if ($this->owner->config()->get('cms_publish_required') !== null) {
-            Deprecation::notice('5.0', 'Use cms_localisation_required instead');
+        $config = $this->owner->config();
+        $inheritanceMode = FluentState::singleton()->getIsFrontend()
```

Comment:
I suggest we detect is_boolean here, and map to the default values. It doesn't have to be a deprecation warning.

E.g. false == none, true === fallback